pkg/execute/kubectl.go

```diff
@@ -0,0 +1,206 @@
+	if !e.resMapping.AllowedKubectlVerbMap[args[0]] {
+		return false
+	}
+
+	return true
+}
+
+// Execute executes kubectl command based on a given args.
+//
+// This method should be called ONLY if:
+// - we are a target cluster,
+// - and Kubectl.CanHandle returned true.
+func (e *Kubectl) Execute(command string, isAuthChannel bool) (string, error) {
+	e.log.Infof("message %s, auth %v", command, isAuthChannel)
+	var (
+		// TODO: https://github.com/kubeshop/botkube/issues/596
+		// use a related config from communicator bindings.
+		kubectlCfg = e.cfg.Executors.GetFirst().Kubectl
+
+		args             = strings.Fields(strings.TrimSpace(command))
+		clusterName      = e.cfg.Settings.ClusterName
+		defaultNamespace = kubectlCfg.DefaultNamespace
+	)
+
+	if !isAuthChannel && kubectlCfg.RestrictAccess {
+		msg := fmt.Sprintf(kubectlNotAuthorizedMsgFmt, clusterName)
+		return e.omitIfIfWeAreNotExplicitlyTargetCluster(command, msg)
+	}
+
+	if !kubectlCfg.Enabled {
+		msg := fmt.Sprintf(kubectlDisabledMsgFmt, clusterName)
+		return e.omitIfIfWeAreNotExplicitlyTargetCluster(command, msg)
+	}
+
+	// Some commands don't have resources specified directly in command. For example:
+	// - kubectl logs foo
+	if !validDebugCommands[args[0]] {
+		// Check if user has access to a given Kubernetes resource
+		// TODO: instead of using config with allowed verbs and commands we simply should use related SA.
+		if len(args) > 1 && !e.matchesAllowedResources(args[1]) {
+			return fmt.Sprintf(kubectlNotAllowedKindMsgFmt, args[1], clusterName), nil
+		}
+	}
+
+	args, executionNs, err := e.ensureNamespaceFlag(args, defaultNamespace)
+	if err != nil {
+		return "", fmt.Errorf("while ensuring Namespace for command execution: %w", err)
+	}
+
+	if !utils.IsNamespaceAllowed(kubectlCfg.Namespaces, executionNs) {
+		return fmt.Sprintf(kubectlNotAllowedNamespaceMsgFmt, executionNs, clusterName), nil
+	}
+
+	finalArgs := e.getFinalArgs(args)
+	out, err := e.runCmdFn(kubectlBinary, finalArgs)
+	if err != nil {
+		err = fmt.Errorf("while executing kubectl command: %w", err)
+		return fmt.Sprintf("Cluster: %s\n%s%s", clusterName, out, err.Error()), err
+	}
+
+	return fmt.Sprintf("Cluster: %s\n%s", clusterName, out), nil
+}
+
+// omitIfIfWeAreNotExplicitlyTargetCluster returns verboseMs if there is explicit '--cluster-name' flag that matches this cluster.
+// It's useful if we want to be more verbose, but we also don't want to spam if we are not the target one.
+func (e *Kubectl) omitIfIfWeAreNotExplicitlyTargetCluster(cmd string, verboseMsg string) (string, error) {
+	if utils.GetClusterNameFromKubectlCmd(cmd) == e.cfg.Settings.ClusterName {
+		return verboseMsg, nil
+	}
+	return "", nil
+}
+
+// TODO: This code was moved from:
+//   https://github.com/kubeshop/botkube/blob/0b99ac480c8e7e93ce721b345ffc54d89019a812/pkg/execute/executor.go#L242-L276
+// Further refactoring in needed. For example, the cluster flag should be removed by a higher logic as it's strictly
```

Comment:
Higher logic sounds a bit odd to me 😄 
```suggestion
// Further refactoring in needed. For example, the cluster flag should be removed by an upper layer as it's strictly
```